frappe/utils/backups.py

```diff
@@ -700,14 +704,16 @@ def decryption_rollback(self):
 				os.remove(self.file_path.rstrip(".gz"))
 			os.rename(self.file_path + ".gpg", self.file_path)
 
+	def __del__(self):
```

Comment:
overriding `__del__` is usually a bad practice. It's an internal method used by garbage collector, what we are doing is more suited for contextmanager.

Reply:
Based on what I understood its mostly safe enough unless the class has cyclic references.

Context managers seem to become a bit messy with the way this code has to be used, will see if I can figure out a clean way to implement